common/Unit.cpp

```diff
@@ -698,8 +704,9 @@ void UnitWSD::DocBrokerDestroy(const std::string& key)
 
                 LOG_TST("Starting test #" << GlobalIndex + 1 << ": "
                                           << GlobalArray[GlobalIndex]->getTestname());
-                if (GlobalWSD)
-                    GlobalWSD->configure(Poco::Util::Application::instance().config());
+                UnitWSD *globalWSD = GlobalWSD;
+                if (globalWSD)
+                    globalWSD->configure(Poco::Util::Application::instance().config());
```

Comment:
We made `GlobalWSD` an atomic only to take a naked copy?!?

If the atomics are needed, this is code is totally broken and nonsensical. How is taking a copy of an atomic-pointer useful?